doc/read-the-docs-site/doc/marconi-as-a-library/tutorials/write-first-indexer.rst

```diff
@@ -1,6 +1,122 @@
+.. _my-first-indexer:
+
 Write your first indexer
 ========================
 
 In this tutorial, we'll show you how to write the simplest possible indexer, and how to expose the query interface of your indexer from a JSON-RPC server.
 
-Coming eventually!
+In order to understand the tutorial, you need to understand:
+
+* Cardano blockchain (specifically `cardano-node`)
+* Haskell (mainly the cardano-api, lenses, sqlite-simple packages)
+* how chain-indexing works in general
+
+The main building blocks of an indexer are:
+
+1) the event, which is the information you want to extract from a block and store it
+2) the instanciation of the backend
```

Comment:
I'm not that happy with the `backend` term here, even if I understand what you meant and what you struggle with, as I also had a hard time differentiating the different indexing steps and naming thing precisely. It's just that `backend` sounds terribly generic.
Would `storage` be (slightly) more accurate? (I'm not happy with it either, so it's more a suggestion than a request)

Reply:
"Storage" is definitely better